### PR TITLE
"output" added to the global user setting

### PR DESCRIPTION
**Pull request by aster94**
Hello, i just modified the output path to work as global (i.e. defined once as the "arduino.path") 
This way the user would need to choose it only once instead of choosing one for every sketch. This could be overwritten in the per sketch setting (i.e. .vscode/arduino.json)

I run a test with F5 and it run well:

debug

honestly I don't know how to do more tests

### CODE REVIEW

File: src/arduino/arduinoSettings.ts [hunk 1]
```diff
@@ -227,4 +235,14 @@ export class ArduinoSettings implements IArduinoSettings {
...
     }
-}
+
+    private async tryGetDefaultOutputPath(): Promise<void> {
+        // 1. Search vscode user settings first.
+        const configValue = VscodeSettings.getInstance().defaultOutputPath;
+        if (!configValue || !configValue.trim()) {
+            //do nothing?
+        } else {
+            this._arduinoPath = configValue;
```
Comment (aster94) [inline src/arduino/arduinoSettings.ts]: typo: `_arduinoPath` should be `_defaultOutputPath`

**Comment by Sneezry** [comment]
Hi @aster94 Thanks for your contribution!

I find you add a new option in config file called `defaultOutputPath`, and the code can set/get the value correctly.

However, I cannot figure out how you use its value. Do you pass it to Arduino IDE CLI or something else? How does the extension overwrites sketch config with the specific value?

BTW, you can just push another commit to fix previous code issue rather than post comments here :-)

**Comment by aster94** [comment]
Honestly i don t know how to procede, but i would be happy to do it
I am quite sure i need to modify the arduino.ts if you can explain me what needs to be done I will do

**Comment by aster94** [comment]
arduino.ts modified, a few testing are required

the defaultOutputPath is overwritten by the local output path

`if (dc.output) {
    const outputPath = path.resolve(ArduinoWorkspace.rootPath, dc.output);
    args.push("--pref", `build.path=${outputPath}`);
} else {
    const msg = "Output path is not specified. Unable to reuse previously compiled files. Upload could be slow. See README.";
    arduinoChannel.warning(msg);
}`

**Comment by aster94** [comment]
I did some errors in this pull request and i deleted it.
please continue on #655

**Pull request closed without merging**